### PR TITLE
Easier saving

### LINKED ISSUE #1813: [Request] Quick Save option
Could we add either a quicksave button or a button on the savegame list to quickly save to the loaded game? 

Sometimes I can't remember which savegame I've been working on. I remember that when it used windows UI to do savegames, it would autofill your filename as the file that you had open, which made things a lot easier. We lost this when it went to the game UI (Although you guys have made it look and operate much better)

Thanks very much

### PR DESCRIPTION
**Pull request by runelaenen**
Replace Save game functionality and add 'save game as' function.

Fixes #1813 I think

### CODE REVIEW

File: data/language/english_uk.txt [hunk 1]
```diff
@@ -885,7 +885,7 @@ STR_0879    :Can't lower land here...
 STR_0880    :Can't raise land here...
 STR_0881    :Object in the way
 STR_0882    :Load Game
-STR_0883    :Save Game
+STR_0883    :Save Game as
```
Comment (AaronVanGeffen) [inline data/language/english_uk.txt]: Why change this string when you're introducing the same one again later? Best to leave the string and just change the string id elsewhere, I think — less hassle for the other languages.

Comment (runelaenen) [inline data/language/english_uk.txt]: I'll change it. I had to choose between changing it in the code or changing it in the language files...

File: data/language/english_uk.txt [hunk 2]
```diff
@@ -3850,3 +3850,4 @@ STR_5508    :Allow loading files with incorrect checksums
 STR_5509    :{SMALLFONT}{BLACK}Allows loading scenarios and saves that have an incorrect checksum, like the scenarios from the demo or damaged saves.
 STR_5510    :Default sound device
 STR_5511    :(UNKNOWN)
+STR_5512    :Save game as
```
Comment (IntelOrca) [inline data/language/english_uk.txt]: I suggest this should be "Save Game As..."

File: src/game.c [hunk 3]
```diff
@@ -990,32 +990,37 @@ static int show_save_game_dialog(char *resultPath)
...
-			game_do_command(0, 1047, 0, -1, GAME_COMMAND_SET_RIDE_APPEARANCE, 0, 0);
-			gfx_invalidate_screen();
-			return 1;
+		strcat(path, gScenarioSaveName);
+		strcat(path, ".sv6");
+
+		SDL_RWops* rw = platform_sdl_rwfromfile(path, "wb+");
+		if (rw != NULL) {
+			scenario_save(rw, 0x80000000);
+			log_error("Saved to %s", gScenarioSaveName);
```
Comment (IntelOrca) [inline src/game.c]: Change this to log_verbose.

File: src/scenario.c [hunk 4]
```diff
@@ -906,9 +907,9 @@ int scenario_save(SDL_RWops* rw, int flags)
 	int viewX, viewY, viewZoom, viewRotation;
 
 	if (flags & 2)
-		log_verbose("saving scenario");
+		log_error("saving scenario");
```
Comment (IntelOrca) [inline src/scenario.c]: Change this to log_verbose.

File: src/scenario.c [hunk 5]
```diff
@@ -906,9 +907,9 @@ int scenario_save(SDL_RWops* rw, int flags)
...
 	else
-		log_verbose("saving game");
+		log_error("saving game");
```
Comment (IntelOrca) [inline src/scenario.c]: Change this to log_verbose.

File: src/localisation/string_ids.h [hunk 6]
```diff
@@ -114,6 +114,7 @@ enum {
 	STR_CANT_RAISE_LAND_HERE = 880,
 
 	STR_LOAD_GAME = 882,
+	STR_SAVE_GAME_AS = 5512,
```
Comment (Gymnasiast) [inline src/localisation/string_ids.h]: String IDs should be sorted numerically, if I'm not mistaken.

File: src/game.c (hunk 3 above)
Comment (runelaenen) [inline src/game.c]: Sorry, I had it changed to test it. I'll change it back.

File: src/localisation/string_ids.h (hunk 6 above)
Comment (runelaenen) [inline src/localisation/string_ids.h]: Why?

Comment (Gymnasiast) [inline src/localisation/string_ids.h]: I remember @IntelOrca saying so and having sorted the list himself a couple of times.

**Pull request merged**